DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Utils/FeatureClassUtils.cs

```diff
@@ -0,0 +1,93 @@
+using ArcGIS.Core.Data;
+using ArcGIS.Core.Data.Exceptions;
+using ArcGIS.Core.Geometry;
+using Objects.GIS;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+
+namespace Speckle.Converters.ArcGIS3.Utils;
+
+public class FeatureClassUtils
+{
+  public void AddFeaturesToFeatureClass(
+    FeatureClass newFeatureClass,
+    VectorLayer target,
+    List<string> fieldAdded,
+    IRawConversion<Base, ArcGIS.Core.Geometry.Geometry> gisGeometryConverter
+  )
+  {
+    newFeatureClass.DeleteRows(new QueryFilter());
+    foreach (GisFeature feat in target.elements.Cast<GisFeature>())
+    {
+      using (RowBuffer rowBuffer = newFeatureClass.CreateRowBuffer())
+      {
+        foreach (string field in fieldAdded)
+        {
+          // try to assign values to writeable fields
+          try
+          {
+            if (feat.attributes is not null)
+            {
+              var value = feat.attributes[field];
+              if (value is not null)
+              {
+                rowBuffer[field] = value.ToString();
+              }
+              else
+              {
+                rowBuffer[field] = null;
+              }
+            }
+          }
+          catch (GeodatabaseFieldException)
+          {
+            // non-editable Field, do nothing
+          }
+        }
+
+        if (feat.geometry != null)
+        {
+          foreach (var geometryPart in feat.geometry)
+          {
+            // TODO: repeat for all geometries, add as Multipart
+            ArcGIS.Core.Geometry.Geometry nativeShape = gisGeometryConverter.RawConvert(geometryPart);
+            rowBuffer[newFeatureClass.GetDefinition().GetShapeField()] = nativeShape;
```

Comment:
are you happy that these cannot return null? that the index/key will be valid?

Reply:
if it's null, it's non-geometry feature, added POC comment